pkg/database/database.go

```diff
@@ -34,3 +36,16 @@ func Factory(databaseType constants.StoreType, store *config.KVStore) (Database,
 		return nil, &DatabaseNotImplementedError{databaseType.Name()}
 	}
 }
+
+func FactoryWithNameAndLatency(databaseType constants.StoreType, name string, latency time.Duration) (Database, error) {
```

Comment:
could Factory method be reused (after proper change if appicable)?

Reply:
Go does not have any ways for method overloading.